src/Common/Contracts/Container.php

```diff
@@ -27,4 +27,79 @@ public function get( $id ) {
 			throw new Not_Bound_Exception( $e->getMessage(), $e->getCode() );
 		}
 	}
+
+	/**
+	 * Overrides the parent method to fire an action when a service provider is registered.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $serviceProviderClass The service provider class name.
+	 * @param string ...$alias             Optional. The alias(es) to register the service provider with.
+	 *
+	 * @return void
+	 *
+	 * @throws \TEC\Common\lucatume\DI52\ContainerException If the provider class is marked as deferred but
+	 *                                                      does not provide a set of deferred registrations.
+	 */
+	public function register( $serviceProviderClass, ...$alias ) {
+		// Register the provider with the parent container.
+		parent::register( $serviceProviderClass, ...$alias );
+
+		/**
+		 * Fires when a service provider is registered by the container.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $serviceProviderClass The service provider class name.
+		 */
+		do_action( 'tec_container_registered_provider', $serviceProviderClass );
```

Comment:
Can we add a
`do_action( 'tec_container_registered_provider_' . $serviceProviderClass );`

To allow for specific targeting w/o having to check every one of these?

Maybe as part of the failed check for `$serviceProviderClass::$registration_action`?

Reply:
PS: do we want to pass/hook alias(es) as well? 🤔

Reply:
Good idea!
I've added a class-specific action, always dispatched whether the custom action is dispatched or not, and included the aliases in the dispatched actions.